crates/consensus/src/verification.rs

```diff
@@ -329,30 +342,29 @@ pub fn full_validation<Provider: HeaderProvider + AccountProvider>(
     let parent = validate_block_regarding_chain(block, &provider)?;
     validate_header_regarding_parent(&parent, &block.header, config)?;
 
-    for transaction in block.body.iter() {
-        validate_transaction_regarding_header(
-            transaction,
-            config,
-            block.header.number,
-            block.header.base_fee_per_gas,
-        )?;
-
-        // NOTE: depending on the need of the stages, recovery could be done in different place.
-        let recovered =
-            transaction.try_ecrecovered().ok_or(Error::TransactionSignerRecoveryError)?;
-
-        let account =
-            provider.basic_account(recovered.signer())?.ok_or(Error::SignerAccountNotExisting)?;
-
-        validate_transaction_regarding_account(transaction, &account)?;
-    }
+    // NOTE: depending on the need of the stages, recovery could be done in different place.
+    let transactions = block
+        .body
+        .iter()
+        .map(|tx| tx.try_ecrecovered().ok_or(Error::TransactionSignerRecoveryError))
+        .collect::<Result<Vec<_>, _>>()?;
```

Comment:
Given this function does sender recovery, which is an expensive operation, let's make sure that we don't use this in the execution stage. Will be useful later when for the builder / Engine API.